### DISCUSSION TITLE
DKIM keys of  mail senders not found

### DISCUSSION THREAD

**Discussion by M-Stenzel** [rspamd/rspamd#5033] [Q&A]
Hi team,

I am running latest version of rspamd.

In the logs I receive sth. like that:

(rspamd_proxy) <aaf896>; proxy; dkim_module_load_key_format: cannot load dkim key /var/lib/rspamd/dkim/gmail.com.dkim.key: cannot stat key file: '/var/lib/rspamd/dkim/gmail.com.dkim.key' No such file or directory

The directory does exist and holds exactly _one_ key, the key of my mail server.

Must be sth. like a configuration error.

Of course I do not have the DKIM key (private part) of **gmail.com**.

What did I do wrong?

Martin.

**Comment by M-Stenzel** [comment]
Do these two parameters have a meaning here?


```
check_pubkey = false;
allow_pubkey_mismatch = true;

```

**Comment by clarkjim2** [comment]
I'll second this as a problem for me as I get the same error using the latest docker-mailserver (14.0). Emails can be sent but are not DKIM signed. Permissions for the rspamd directory and the DKIM key are set to _rspamd:_rspamd. 

I also have the same pubkey settings. 

~J

**Reply by clarkjim2** [reply]
To get DKIM signing to work, I had to disable Rspamd and enable OpenDKIM. 

I spent a lot of time trying to get Rspamd to work by adjusting permissions, moving the key to /tmp (see #3343), tested both RSPAMD_CHECK_AUTHENTICATED =0 and =1 (#3665), and a bunch of other things that never eliminated the 'cannot stat key file' error.
